driver/nodeserver.go

```diff
@@ -83,10 +85,17 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 
 	log.Info("Volume registered for management")
 
-	if err := wait.PollUntil(time.Second, func() (done bool, err error) {
-		return ns.manager.IsVolumeReady(req.GetVolumeId()), nil
-	}, ctx.Done()); err != nil {
-		return nil, err
+	// Only wait for the volume to be ready if it is in a state of 'ready to request'
+	// already. This allows implementors to defer actually requesting certificates
+	// until later in the pod lifecycle (e.g. after CNI has run & an IP address has been
+	// allocated, if a user wants to embed pod IPs into their requests).
+	isReadyToRequest := ns.manager.IsVolumeReadyToRequest(req.GetVolumeId())
+	if isReadyToRequest || !ns.continueOnNotReady {
```

Comment:
Could we simplify the logic here that, wait for volumeReady only if `continueOnNotReady` is false ?
Here, if a driver use default `AlwaysReadyToRequest` func, `isReadyToRequest` may always return `true`. In this case, `continueOnNotReady` becomes useless ?

Reply:
Correct - the logic is that if we _are_ ready to request straight away, we wait/block. This was intentional so that we only skip waiting *if* we aren't 'ready' to wait (and it's been configured to skip in these cases)